packages/admin-next/dashboard/src/routes/locations/location-fulfillment-providers/components/edit-fulfillment-providers-form/edit-fulfillment-providers-form.tsx

```diff
@@ -0,0 +1,198 @@
+import { useDataTable } from "../../../../../hooks/use-data-table"
+
+type LocationEditFulfillmentProvidersFormProps = {
+  location: HttpTypes.AdminStockLocation
+}
+
+const EditFulfillmentProvidersFormSchema = zod.object({
+  fulfillment_providers: zod.array(zod.string()).optional(),
+})
+
+const PAGE_SIZE = 50
+
+export const LocationEditFulfillmentProvidersForm = ({
+  location,
+}: LocationEditFulfillmentProvidersFormProps) => {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+
+  const form = useForm<zod.infer<typeof EditFulfillmentProvidersFormSchema>>({
+    defaultValues: {
+      fulfillment_providers:
+        location.fulfillment_providers?.map((fp) => fp.id) ?? [],
+    },
+    resolver: zodResolver(EditFulfillmentProvidersFormSchema),
+  })
+
+  const { setValue } = form
+
+  const initialState =
+    location.fulfillment_providers?.reduce((acc, curr) => {
+      acc[curr.id] = true
+      return acc
+    }, {} as RowSelectionState) ?? {}
+
+  const [rowSelection, setRowSelection] =
+    useState<RowSelectionState>(initialState)
+
+  useEffect(() => {
+    const ids = Object.keys(rowSelection)
+    setValue("fulfillment_providers", ids, {
+      shouldDirty: true,
+      shouldTouch: true,
+    })
+  }, [rowSelection, setValue])
+
+  const { searchParams, raw } = useFulfillmentProvidersTableQuery({
+    pageSize: PAGE_SIZE,
+  })
+
+  const { fulfillment_providers, count, isLoading, isError, error } =
+    useFulfillmentProviders(
+      { ...searchParams },
+      { placeholderData: keepPreviousData }
+    )
+
+  const filters = useDateTableFilters()
+  const columns = useColumns()
+
+  const { table } = useDataTable({
+    data: fulfillment_providers ?? [],
+    columns,
+    count,
+    enablePagination: true,
+    enableRowSelection: true,
+    rowSelection: {
+      state: rowSelection,
+      updater: setRowSelection,
+    },
+    getRowId: (row) => row.id,
+    pageSize: PAGE_SIZE,
+  })
+
+  const { mutateAsync, isPending: isMutating } =
+    useUpdateStockLocationFulfillmentProviders(location.id)
+
+  const handleSubmit = form.handleSubmit(async (data) => {
+    const originalIds = location.fulfillment_providers?.map((sc) => sc.id)
+
+    const arr = data.fulfillment_providers ?? []
+
+    await mutateAsync(
+      {
+        add: arr.filter((i) => !originalIds?.includes(i)),
+        remove: originalIds?.filter((i) => !arr.includes(i)),
+      },
+      {
+        onSuccess: () => {
+          toast.success(t("stockLocations.fulfillmentProviders.successToast"))
+          handleSuccess(`/settings/locations/${location.id}`)
+        },
+        onError: (e) => {
+          toast.error(e.message)
+        },
+      }
+    )
+  })
+
+  if (isError) {
+    throw error
+  }
+
+  return (
+    <RouteFocusModal.Form form={form}>
+      <div className="flex h-full flex-col overflow-hidden">
+        <RouteFocusModal.Header>
```

Comment:
suggestion: believe we started moving header actions to footer. Iinstead this use ` <RouteFocusModal.Footer>` after `<RouteFocusModal.Body>`.